core/drivers/imx_i2c.c

```diff
@@ -0,0 +1,518 @@
+// SPDX-License-Identifier: GPL-2.0+
```

Comment:
Unfortunately we can't accept any GPL licensed code in OP-TEE OS :-(
Only BSD license and compatible are OK.

Reply:
ah, ok.. let me see if there is anything I can do about it but from past experiences I dont feel too optimistic.  I'll post tomorrow the linux driver then. thanks for the quick feedback!

Reply:
> ah, ok.. let me see if there is anything I can do about it but from past experiences I dont feel too optimistic.

The only options I see are, either all the authors give their authorization to change the license, or the code needs to be rewritten from scratch.

> I'll post tomorrow the linux driver then. thanks for the quick feedback!

You're welcome!

Reply:
I ended up rewriting the driver following up the manual (is manly just a read/write transfer operation...so not many different ways of doing it after all). let me know what you think please.

Reply:
> I ended up rewriting the driver following up the manual

That's good, makes it totally acceptable here as far as I can tell.
After a very quick look it looks good to me:

`Acked-by: Jerome Forissier <jerome@forissier.org>`

I will leave some time for others and perhaps i.MX reviewers to comment further.
Also, would you want to add yourself as a reviewer in MAINTAINERS perhaps? IMO it would be a good idea since (1) you wrote the code and (2) this doesn't live under the `plat-imx` directory.

Reply:
thanks @jforissier I added a change to the MAINTAINERS file as a separate commit.